Apps/WebClient/src/Server/Controllers/CommentController.cs

```diff
@@ -0,0 +1,186 @@
+            AuthorizationResult isAuthorized = await this.authorizationService
+                .AuthorizeAsync(user, userHdid, PolicyNameConstants.UserIsPatient)
+                .ConfigureAwait(true);
+            if (!isAuthorized.Succeeded)
+            {
+                return new ForbidResult();
+            }
+
+            if (comment == null)
+            {
+                return new BadRequestResult();
+            }
+
+            if (comment.HdId != userHdid)
+            {
+                return new ForbidResult();
+            }
+
+            comment.UpdatedBy = userHdid;
+            RequestResult<Comment> result = this.commentService.Update(comment);
+            return new JsonResult(result);
+        }
+
+        /// <summary>
+        /// Deletes a Comment from the database.
+        /// </summary>
+        /// <returns>The deleted Comment wrapped in a RequestResult.</returns>
+        /// <param name="comment">The comment to be deleted.</param>
+        /// <response code="200">The note was deleted.</response>
+        /// <response code="401">The client must authenticate itself to get the requested response.</response>
+        /// <response code="403">The client does not have access rights to the content; that is, it is unauthorized, so the server is refusing to give the requested resource. Unlike 401, the client's identity is known to the server.</response>
+        [HttpDelete]
+        [Authorize(Policy = "PatientOnly")]
+        public async Task<IActionResult> Delete([FromBody] Comment comment)
+        {
+            // Validate the hdid to be a patient.
+            ClaimsPrincipal user = this.httpContextAccessor.HttpContext.User;
+            string userHdid = user.FindFirst("hdid").Value;
+            AuthorizationResult isAuthorized = await this.authorizationService
+                .AuthorizeAsync(user, userHdid, PolicyNameConstants.UserIsPatient)
+                .ConfigureAwait(true);
+            if (!isAuthorized.Succeeded)
+            {
+                return new ForbidResult();
+            }
+
+            RequestResult<Comment> result = this.commentService.Delete(comment);
+            return new JsonResult(result);
+        }
+
+        /// <summary>
+        /// Gets all comments for the authorized user and event id.
+        /// </summary>
+        /// <param name="parentEntryId">The parent entry id.</param>
+        /// <returns>The list of comments wrapped in a request result.</returns>
+        /// <response code="200">Returns the list of comments.</response>
+        /// <response code="401">The client must authenticate itself to get the requested response.</response>
+        /// <response code="403">The client does not have access rights to the content; that is, it is unauthorized, so the server is refusing to give the requested resource. Unlike 401, the client's identity is known to the server.</response>
+        [HttpGet]
+        [Route("{parentEntryId}")]
```

Comment:
I am not 100% this is restful `Comment/{parent_id}` since it might look similar to the comment id. How about `Comment/?parentId={parent_id}